gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/KillDagThread.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.modules.orchestration;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.instrumented.Instrumented;
+import org.apache.gobblin.metrics.event.TimingEvent;
+import org.apache.gobblin.service.ExecutionStatus;
+import org.apache.gobblin.service.modules.flowgraph.Dag;
+import org.apache.gobblin.service.modules.spec.JobExecutionPlan;
+import org.apache.gobblin.service.monitoring.JobStatus;
+
+import static org.apache.gobblin.service.ExecutionStatus.CANCELLED;
+import static org.apache.gobblin.service.ExecutionStatus.ORCHESTRATED;
+import static org.apache.gobblin.service.ExecutionStatus.valueOf;
+
+
+@Slf4j
+public class KillDagThread implements Runnable {
+  private final Long defaultJobStartSlaTimeMillis;
+  NewDagManager dagManager;
+  public KillDagThread(Long defaultJobStartSlaTimeMillis, NewDagManager newDagManager) {
+    this.defaultJobStartSlaTimeMillis = defaultJobStartSlaTimeMillis;
+    this.dagManager = newDagManager;
+  }
+
+  @Override
+  public void run() {
+    for (Dag.DagNode<JobExecutionPlan> node : this.dagManager.getJobToDag().keySet()) {
+      boolean flowKilled = enforceFlowStartDeadline(node);
+      boolean jobKilled = false;
+
+      if (!flowKilled) {
+        JobStatus jobStatus = pollJobStatus(node);
+        try {
+          jobKilled = enforceJobCompletionDeadline(node, jobStatus);
```

Comment:
do we try to enforce the SLA kills first even when we receive an API request? I was not aware this is the norm. Is this not supposed to kill the dag because of api request or its used differently?